File: BiteClub/Models/OrderLocation.swift
import Foundation

struct OrderLocation: Codable, Identifiable {
    var id: UUID
    var orderId: UUID
    var locationId: UUID
    var createdAt: Date
    var updatedAt: Date
    
    var order: Order?
    var location: Location?
    var orderItems: [OrderItem]?
    
    enum CodingKeys: String, CodingKey {
        case id = "id"
        case orderId = "order_id"
        case locationId = "location_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case order = "order"
        case location = "location"
        case orderItems = "orderItems"
    }
}

File: BiteRunr/shared/src/commonMain/kotlin/org/biterunr_studios/biterunr/Models/OrderLocation.kt
package org.biterunr_studios.biterunr.Models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.datetime.Instant

@Serializable
data class OrderLocation(
    @SerialName("id")
    val id: String,

    @SerialName("order_id")
    val orderId: String,

    @SerialName("location_id")
    val locationId: String,

    @SerialName("created_at")
    val createdAt: Instant,

    @SerialName("updated_at")
    val updatedAt: Instant,

    @SerialName("order")
    val order: Order? = null,

    @SerialName("location")
    val location: Location? = null,

    @SerialName("order_items")
    val orderItems: List<OrderItem>? = null
)
